### DISCUSSION TITLE
How to set button loading style on form submit rather than just on click

### DISCUSSION THREAD

**Discussion by markdynamo6** [filamentphp/filament#10754] [Help] [answered]
I have the following simple form:

```php
public function form(Form $form): Form
{
        return $form
            ->schema([
                TextInput::make('some_text'),
                Actions::make([
                    Action::make('doSomething')
                        ->label('Submit')
                        ->action(fn () => $this->create())
                        ->icon('heroicon-m-magnifying-glass')
                ])
            ])
            ->statePath('data')
            ->model(Member::class);
}
```

When I click the button, the loading spinner replaces the magnifying glass as expected and the `some_text` submits.
When the text input has focus and I just hit the return key, the data also submits, but this time the loading spinner does not replace the maginifying glass.

How do I get the spinner to show on form submit, rather than just on button click?

**Comment by leandrocfe** [comment, marked as answer]
```php
Action::make('doSomething')
    ->submit('create')
```
?

**Reply by markdynamo6** [reply]
That's awesome. I knew it would be something simple.
Thanks
